iceoryx2-bb/container/tests/flatmap_tests.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use iceoryx2_bb_container::flatmap::{FixedSizeFlatMap, FlatMapError};
+use iceoryx2_bb_elementary_traits::placement_default::PlacementDefault;
+use iceoryx2_bb_elementary_traits::zero_copy_send::ZeroCopySend;
+use iceoryx2_bb_testing::assert_that;
+use iceoryx2_bb_testing::lifetime_tracker::LifetimeTracker;
+use iceoryx2_bb_testing::memory::RawMemory;
+
+mod flat_map {
+    use super::*;
+
+    const CAPACITY: usize = 100;
+
+    // used to check if drop is called for the key-value pairs if the map is dropped
+    struct RelocLifetimeTracker(LifetimeTracker);
```

Comment:
Does it make sense to provide the `RelocatableLifetimeTracker` in some central place again like `iceoryx2_bb_testing`?